### PR TITLE
DP-21041 External data resource content type

### PR DESCRIPTION
**Pull request by ygannett**
**Description:**
Create a new content type, External data resource for data tab.

- Content-type name:  External data resource
- Description: This is used to designate an external page as a state data source so that it will be linked to the state's open data listing view.
- Category:  Other
- Fields:
   - **Title** 
      - No modification. Set up as its default setting.
   - **URL**
      - No link text
      - External link only
      - Custom help text for external link.
      - Required
   - **Description**
      - Max char. 500 (based on the current search results description's 1 line contains 100 char.) - this can be changed.
      - Simple text, no RTE
      - Required
   - **Organization (one or more)**
      - Multiple inserts
      - Required
   - **Data Type (data catalog, dataset, data report) (same field used on info details)**
      - Required
   - **Data resource type** (same field used on info details)
      - Same options as one in info details
      - Displayed when the Data Type field has _data report_.
   - **Data format (same field used on info details)**
      - Displayed when the Data Type field has the Data resource value.
      - Required when the Data Type field has the Data resource value.
      - [NOTE] Currently, the Data resource option is not mapped to the Data resource type (field_data_resource_type) as the mapping with done in the mass meta tag module to output as meta data in a page.

Create a new role, **External data resource manager**,  for the content type.

Update the data type field help text in **info. details**, **service details**, and **service**  from "Data resource: A report, data visualization or map." to "Data resource: A report.".

Format the View of the content based on one for Document:
view-dataset

The Data resource type is displayed only when the Data Type has "data report":
view-data-resource

**Jira:** (Skip unless you are MA staff)
https://jira.mass.gov/browse/DP-21041

**To Test:**
- [ ] Click the **Add content** button, then click the **Other** to reach the **External data resource**.
- [ ] Create a new **External data resource**.
- [ ] In the edit page, check the field names, types, descriptions, help text, and etc. are set up as specked above.

**Screenshots/GIFs:**
add-content

edit-external-data

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: docroot/modules/custom/mass_utility/mass_utility.module [hunk 1]
```diff
@@ -131,7 +131,7 @@ function mass_utility_form_node_form_alter(&$form, FormStateInterface $form_stat
       '#type' => 'details',
       '#title' => t('SiteImprove Groups'),
       '#group' => 'advanced',
-      '#weight' => 90,
+      '#weight' => 90
```
Comment (tom-fleming) [inline docroot/modules/custom/mass_utility/mass_utility.module]:
```suggestion
      '#weight' => 90,
```
This is a PHP coding standard recommendation

File: docroot/modules/custom/mass_utility/mass_utility.module [hunk 2]
```diff
@@ -472,7 +472,14 @@ function mass_utility_field_widget_form_alter(&$element, FormStateInterface $for
     }
     else {
       // Set link field descriptions for all other link fields.
-      $element['uri']['#description'] = t('Start typing to choose an existing page on Mass.gov, or enter a complete URL to another site. If you want to add an external link, it must begin with "https://" or "http://".');
+
+      // Override description of the URL field in External data resource content type.
+      if ($field_name == 'field_externaldata_url') {
+        $element['uri']['#description'] = t('Enter a complete URL, it must begin with "https://" or "http://".');
```
Comment (tom-fleming) [inline docroot/modules/custom/mass_utility/mass_utility.module]:
```suggestion
        $element['uri']['#description'] = t('Enter a complete URL. It must begin with "https://" or "http://".');
```

File: docroot/themes/custom/mass_theme/templates/content/node--external-data-resource.html.twig [hunk 3]
```diff
@@ -0,0 +1,55 @@
+{#
+/**
+ * @file
+ * Theme override to display the External Data Resource content type.
+ * Internal use only for reveiw input data.
```
Comment (tom-fleming) [inline docroot/themes/custom/mass_theme/templates/content/node--external-data-resource.html.twig]:
```suggestion
 * Internal use only for review input data.
```

File: docroot/modules/custom/mass_utility/mass_utility.module (hunk 1 above)
Comment (ygannett) [reply docroot/modules/custom/mass_utility/mass_utility.module]: Thank you.  That's corrected.

**Pull request merged**